### PR TITLE
Bugfix/atr 741 dev px1007 inheritdoc support

### PR DESCRIPTION
**Pull request by SENya1990**
Big rework of PX1007 diagnostic. Overview:
- added analysis support for <inheritdoc> XML doc tag:
   - For non-projection DACs, DAC extensions and DAC field properties its ok to just have one instead of a _summary_ tag. Its format is not checked

   - For DAC field properties in projection DACs and extensions to projection DACs there is additional check [ATR-753]. If _inheritdoc_ tags (there can be more than 1) on **mapped** projection DAC properties do not point to the original DAC property then it will be reported as error. 
   
   - Mapped properties are determined from the `BqlField` or `BqlTable` values specified in attributes. Not mapped properties (unbound or DB Calced) are processed as normal DAC properties
   
   - It should look like this:
```csharp
        /// <inheritdoc cref="OriginalDac.BranchID"/>      //Reuse the original DAC field property description
        [Branch(BqlField = typeof(OriginalDac.branchID))]  //Note the BqlField property initialization expression
        public virtual Int32? BranchID { get; set; }
```
   
- added code fix that will generate correctly <inheritdoc> XML doc tag for a mapped projection DAC field property. @EAndrosova by using this code fix and VS "Fix All" feature Doc Team can instantly generate documentation for entire projection DAC or for the whole file/project/solution with such DACs.

- PX1007 did not support empty xml comments like `<summary/>` at all. Added such support, reworked code fix with  _summary_ tag generation

- removed default diagnostic justification `to be documented later` from PX1007 suppression comment (ATR-754). Suppression should not be used to exclude API from the documentation, <exclude> tag should be used instead. 

- `PXHidden` attribute no longer can be used to suppress PX1007 on DAC extensions (ATR-749)

- PX1007 no longer requires documentation from system or well-known DAC properties. The list of excluded properties was extended with Selected, Attributes, GroupMask. Forbidden reserved DAC properties are also excluded: CompanyID, CompanyMask, DeletedDatabaseRecord

- Split the code fix code into three separate code actions - add _summary_ tag, add _inheritdoc_ tag, add _exclude_ tag. Fixed badly formatted indent of generated tags in several cases (badly formatted code, existing other doc tags, no empty lines between BQL field and property)
  
- added helpers developed during the work on .Net Core compatibility analyzer - uniform overrides chain retrieval and some others. 

- refactoring, nullability annotations, debugging experience enhancements

- added a lot of unit tests to cover different scenarios possible with XML doc comments

### CODE REVIEW

File: src/Acuminator/Acuminator.Analyzers/DiagnosticsShortName.resx [hunk 1]
```diff
@@ -387,4 +387,7 @@
   <data name="PX1096" xml:space="preserve">
     <value>PXOverrideSignatureMismatch</value>
   </data>
+  <data name="PX1007IncorrectProjectionDacFieldComment" xml:space="preserve">
```
Comment (SENya1990) [inline src/Acuminator/Acuminator.Analyzers/DiagnosticsShortName.resx:390]: @EAndrosova  please review the short name for new PX1007 diagnostic on projection DAC property incorrect comment

File: src/Acuminator/Acuminator.Analyzers/Resources.resx [hunk 2]
```diff
@@ -159,11 +159,11 @@
   <data name="PX1006Title" xml:space="preserve">
     <value>This order of view declarations will cause creation of one cache instance for the following DACs: {0} and {1}</value>
   </data>
-  <data name="PX1007FixExcludeClass" xml:space="preserve">
-    <value>Exclude the class from the API Reference</value>
+  <data name="PX1007FixAddExcludeTag" xml:space="preserve">
+    <value>Add an "exclude" tag to exclude this API from the API Reference</value>
```
Comment (SENya1990) [inline src/Acuminator/Acuminator.Analyzers/Resources.resx:163]: @EAndrosova  please review the changes in the code fix messages

File: src/Acuminator/Acuminator.Analyzers/Resources.resx [hunk 3]
```diff
@@ -480,8 +480,8 @@
   <data name="PX1043TitleRowPersistedNonISV" xml:space="preserve">
     <value>Only the methods of the PXCache.Persist family can be used to save changes to the database from a RowPersisted event handler</value>
   </data>
-  <data name="PX1007FixAddDescription" xml:space="preserve">
-    <value>Add a description</value>
+  <data name="PX1007FixAddSummaryTag" xml:space="preserve">
+    <value>Add a "summary" tag with description generated from the API name</value>
```
Comment (SENya1990) [inline src/Acuminator/Acuminator.Analyzers/Resources.resx:484]: @EAndrosova  please review the changes in the code fix message

File: src/Acuminator/Acuminator.Analyzers/Resources.resx [hunk 4]
```diff
@@ -159,11 +159,11 @@
...
   </data>
-  <data name="PX1007FixExcludeClass" xml:space="preserve">
-    <value>Exclude the class from the API Reference</value>
+  <data name="PX1007FixAddExcludeTag" xml:space="preserve">
+    <value>Add an "exclude" tag to exclude this API from the API Reference</value>
   </data>
-  <data name="PX1007Title" xml:space="preserve">
-    <value>The DAC, DAC extension, or DAC property should have a description in the summary XML tag</value>
+  <data name="PX1007Title_MissingDescription" xml:space="preserve">
+    <value>The DAC, DAC extension, or DAC property should have a description in the "summary" XML tag or "inheritdoc" XML tag</value>
```
Comment (SENya1990) [inline src/Acuminator/Acuminator.Analyzers/Resources.resx:166]: @EAndrosova  please review the changes in the diagnostic message

File: src/Acuminator/Acuminator.Analyzers/Resources.resx [hunk 5]
```diff
@@ -673,4 +673,15 @@
   <data name="PX1096Title" xml:space="preserve">
     <value>The signature of the method with the PXOverride attribute must match the overriden method</value>
   </data>
+  <data name="PX1007Title_IncorrectProjectionDacFieldDescription" xml:space="preserve">
```
Comment (SENya1990) [inline src/Acuminator/Acuminator.Analyzers/Resources.resx:676]: @EAndrosova please review new diagnostic and code fix messages

File: src/Acuminator/Acuminator.Vsix/Resources/VSIXResource.resx [hunk 6]
```diff
@@ -289,7 +289,7 @@
     <value>Loading the Acuminator package</value>
   </data>
   <data name="Setting_CodeAnalysis_PX1007DiagnosticEnabled_Description" xml:space="preserve">
-    <value>Display the PX1007 diagnostic, which is shown for the public classes, interfaces, structs, enumerations, delegates, and DAC properties that do not have XML comments.</value>
+    <value>Enable Acuminator PX1007 diagnostic that checks DACs, DAC extensions and DAC field properties for the presence of valid XML documentation comments.</value>
```
Comment (SENya1990) [inline src/Acuminator/Acuminator.Vsix/Resources/VSIXResource.resx:292]: @EAndrosova please review changes in the VS setting description that is used to enable PX1007 diagnostic

**Review by dropsonic — APPROVED** [review]

**Review by EAndrosova — APPROVED** [review]

**Pull request merged**